chiselFrontend/src/main/scala/chisel3/core/UserModule.scala

```diff
@@ -97,10 +103,6 @@ abstract class ImplicitModule(implicit moduleCompileOptions: CompileOptions)
   private[core] override def initializeInParent() {
     implicit val sourceInfo = UnlocatableSourceInfo
 
-    for (port <- getModulePorts) {
```

Comment:
This just eliminates redundant/duplicated code. explicitInvalidate is dealt with in the parent class.

Reply:
Ok, and having the DefInvalidate in the parent class handles both cases where the parent doesn't drive inputs and where the instance doesn't drive outputs?

Reply:
Hmmm. I'm not sure I understand this question. I meant the logic in ImplicitModule is a duplicate of logic in the parent class (UserModule.generateComponent()). If ImplicitModule overrode generateComponent() and didn't invoke its parent's implementation, then we'd need the duplicate. Am I missing something more subtle here?

Reply:
So what happens if a module doesn't assign anything to its output? The DefInvalid shouldn't be set in the parent class since it would be trying to write the output of a child module.